homeassistant/components/squeezebox/media_player.py

```diff
@@ -451,6 +457,66 @@ async def async_play_media(
             )
             media_id = play_item.url
 
+        if announce:
+            extra = kwargs.get(ATTR_MEDIA_EXTRA, {})
+
+            _announce_volume: str | None = extra.get("announce_volume", None)
+
+            if not media_id.startswith(SQUEEZEBOX_SOURCE_STRINGS):
+                # do not process special squeezebox "source" media ids
+                media_id = async_process_play_media_url(self.hass, media_id)
```

Comment:
Right now, this code seems to assume that the `media_type` will be MediaType.MUSIC but it could be MediaType.PLAYLIST. 

I'll add `announce` as a parameter to the existing `async_load_url` and `async_load_playlist` functions and then we can keep the existing logic for loading the media.

Reply:
I couldn't think of a scenario where you'd actually want to use a playlist as an announcement.  My other concern was we have to wait for the announcement to stop so we can restore the old playlist, and I put a timeout of 120 secs on that.  We could have no timeout of course, but that felt a bit risky when we might be getting TTS input for example.  No reason to have it really, but couldn't think of a safe way of ending waiting for the stop state otherwise - we could have a process waiting for a long time.

Reply:
Although, we could add another optional extra setting to announce to allow you to override the timeout easily enough - that would probably be a good thing to do anyway

Reply:
As mentioned on discord, playlist causes problems and IMHO doesn't make sense, so I've trapped it with an error.